packages/scandipwa/src/component/ProductCompare/ProductCompare.container.js

```diff
@@ -90,12 +91,31 @@ export class ProductCompareContainer extends PureComponent {
             ...rest,
             attribute_code,
             attribute_values: products.map(({ attributes }) => {
-                const { attribute_value = null } = attributes[attribute_code] || {};
-                return attribute_value;
+                const {
+                    attribute_value = null,
+                    attribute_options = {},
+                    attribute_type
+                } = attributes[attribute_code] || {};
+
+                return attribute_type === ATTR_MULTISELECT
+                    ? this.getMultiSelectAttributeValue(attribute_value, attribute_options)
+                    : this.getAttributeValue(attribute_value, attribute_options);
             })
         }));
     }
 
+    getAttributeValue(value, options) {
+        return options[value]?.label || value || null;
+    }
+
+    getMultiSelectAttributeValue(value, options) {
+        return value
+            .split(/,\s*/)
+            .map((v) => this.getAttributeValue(v, options))
+            .filter((v) => !!v)
```

Comment:
Alternatively, you may use `something.filter(x => x)` or `something.filter(Boolean)`, but this is fine as well.